MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -319,14 +322,16 @@ - (void)onProcessingLog:(id<MSLog>)log withPriority:(MSPriority)priority {
   @synchronized(self) {
     NSData *serializedLog = [NSKeyedArchiver archivedDataWithRootObject:log];
     if (serializedLog && (serializedLog.length > 0)) {
-      if (self.bufferIndex > (ms_crashes_log_buffer_size - 1)) {
-        self.bufferIndex = 0;
+      if ([[self.bufferIndex objectForKey:@(priority)] integerValue] > (ms_crashes_log_buffer_size - 1)) {
+        // Reset the counter to 0 for a priority.
```

Comment:
I would write `self.bufferIndex[@(priority)].integerValue` here, visually simpler to parse IMO